Refactor and extend booking functionality:

iftikhta opened the pull request:
- Consolidate data handling for cottages and boats in cottages.tsx
- Update CottageDetails component for improved data retrieval
- Enhance BookableListItem to handle generic Bookable items
- Introduce BoatDetails component for boat-specific details
- Define Bookable interface and Boat interface with relevant properties
- Add boats data with booking details and availability
- Update package dependencies for type definitions
- #13

iftikhta committed:
Refactor and extend booking functionality:
- Consolidate data handling for cottages and boats in cottages.tsx
- Update CottageDetails component for improved data retrieval
- Enhance BookableListItem to handle generic Bookable items
- Introduce BoatDetails component for boat-specific details
- Define Bookable interface and Boat interface with relevant properties
- Add boats data with booking details and availability
- Update package dependencies for type definitions

iftikhta committed:
corrected boat var name for details page

iftikhta committed:
fixed index setup, commented out cottages.tsx for now

iftikhta committed:
removed cottages.tsx for now, made baot/cottages data ids unique and fixed BookableListItems.tsx for RN mobile

nfgrep commented on app/(tabs)/_layout.tsx:
```diff
@@ -13,7 +13,7 @@ export default function TabLayout() {
   return (
     <Tabs
-      initialRouteName='cottages'
+      initialRouteName="index"
```
This is the default, we can remove this line

nfgrep commented on app/(tabs)/index.tsx:
```diff
@@ -1,5 +1,30 @@
+
+// export default function Index() {
+//   return <Redirect href="/cottages" />;
+// }
```
```suggestion

```

nfgrep reviewed (approved):
Some nits but great overall

iftikhta committed:
Update app/(tabs)/index.tsx

duno why i left this in there, think i was just speed running cahnges but also trying to call it a day

Co-authored-by: Nathan FG <38052787+nfgrep@users.noreply.github.com>

iftikhta committed:
removed index initial route entry

iftikhta committed:
Merge branch 'task/bookable-items' of https://github.com/nfgrep/portage into task/bookable-items
